app/scripts/controllers/permissions/background-api.js

```diff
@@ -38,23 +39,47 @@ export function getPermissionBackgroundApiMethods({
   };
 
   // To add more than one account when already connected to the dapp
-  const addMoreAccounts = (origin, accounts) => {
+  const addMoreAccounts = (origin, addresses) => {
     const caip25Caveat = getCaip25Caveat(origin);
     if (!caip25Caveat) {
       throw new Error(
         `Cannot add account permissions for origin "${origin}": no permission currently exists for this origin.`,
       );
     }
 
-    const ethAccounts = getEthAccounts(caip25Caveat.value);
+    const internalAccounts = addresses.map((address) => {
+      return accountsController.getAccountByAddress(address);
+    });
 
-    const updatedEthAccounts = Array.from(
-      new Set([...ethAccounts, ...accounts]),
-    );
+    const caipAccountIds = internalAccounts.map((internalAccount) => {
+      const { namespace, reference } = parseCaipChainId(
+        internalAccount.scopes[0],
+      );
```

Comment:
not iterating over every single scope is okay for now since `setPermittedAccounts` will set the account on each matching namespace anyway, but this doesn't make me happy still